src/Library/demos/Drag and Drop/main.js

```diff
@@ -0,0 +1,60 @@
+import Gtk from "gi://Gtk";
+import Adw from "gi://Adw";
+import Gdk from "gi://Gdk";
+import GObject from "gi://GObject";
+
+const list = workbench.builder.get_object("list");
+
+// Calculate the length of our ListBox
+let last_row = list.get_last_child();
+let list_length = last_row.get_index() + 1;
+
+// Create a Dragsource and DropTarget for each row
+for (let i = 0; i < list_length; i++) {
+  let row = list.get_row_at_index(i);
+  row.selectable = false;
+  let _drag_x;
+  let _drag_y;
+
+  let allocation;
+  const drag = new Gtk.DragSource();
+  const drop = Gtk.DropTarget.new(row.constructor.$gtype, Gdk.DragAction.COPY);
+
+  row.add_controller(drag);
+  row.add_controller(drop);
+
+  // Drag controller
+  drag.connect("prepare", (source, _x, _y) => {
+    _drag_x = _x;
+    _drag_y = _y;
+
+    allocation = row.get_allocation();
+    let value = new GObject.Value();
+    value.init(GObject.TYPE_OBJECT);
```

Comment:
```suggestion
    value.init(Gtk.ListBoxRow);
```

This should be as specific as possible, and we know the widget must be a `Gtk.ListBoxRow` because it's a child of a `Gtk.ListBox`.